### PR TITLE
clear cache on address controller

### LINKED ISSUE #28983: FO - Bad wording on footer with "Add a first address" and "Addresses"
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

To have the good wording for Addresses on "Your account" block on footer, we need to clear cache on the BO after every action about Addresses part on customer account

Two cases of the bug

1. Create a new account without any addresses, the "Add a first address" wording is well displayed. If we add an address to this account, the wording doesn't change:

https://user-images.githubusercontent.com/2854834/177586591-7fd16ebc-1cf4-4695-928e-877475307b63.mp4

2. Create a new account without any addresses, , the "Add a first address" wording is well displayed. Sign out and connect with an existing customer account with registered addresses, the wording doesn't change:

https://user-images.githubusercontent.com/2854834/177586683-a480a5e0-e91d-402c-ae80-64210c12f458.mp4

Tested and error on 8.0.0 version.
On 1.7.8.6 / 1.7.7.8, this feature is not available.

### Expected behavior

The wording on this block should be modified: 
- no addresses: "Add a first address" must be displayed
- when the first address is added: "Addresses" must be displayed 
- remove all addresses of the customer account: "Add a first address" must be displayed

### Steps to reproduce

**Case 1: create only new account with addresses**
1. Go to FO > Sign in on header > Click on "No account? Create one here" link
2. Fill the form and save
3. Go to footer, "Your account" block > see the good wording "Add a first address"
4. Click on this link > Fill the form with your address and save
5. Go back to the footer > See error, the wording has not changed to "Addresses"
6. Go to BO > Advanced Parameters > Performance > Click on "Clear cache" button on the top of the page > Wait the successful message
7. Go back to FO > Refresh the page > The wording on the footer has changed to "Addresses"

**Case 2: create new account with addresses, sign out and sign in with existing account**

1. Go to FO > Sign in on header > Click on "No account? Create one here" link
2. Fill the form and save
3. Go to footer, "Your account" block > see the good wording "Add a first address"
4. Sign out and sign in again with an existing account with addresses already created, for example "pub@prestashop.com"
5. Go back to the footer > See error, the wording is still "Add a first address" instead of "Addresses"
6. Go to BO > Advanced Parameters > Performance > Click on "Clear cache" button on the top of the page > Wait the successful message
7. Go back to FO > Refresh the page > The wording on the footer has changed to "Addresses"

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by LouisAUTHIE**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Fixes wording problem switching from singular to plural on addresses by clearing cache on addresses updating.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28983 
| Related PRs       | n/a
| How to test?      | Like on the issue testing protocol
| Possible impacts? | Clears the cache on addresses manipulations, so it can slow a bit the reloading of the page at that time

### CODE REVIEW

**Comment by ghost** [comment]
Hello, 

How about adding a key to the cache name directly in the module when a customer have an addresse?

https://github.com/PrestaShop/ps_customeraccountlinks/blob/dev/ps_customeraccountlinks.php#L89

**Comment by LouisAUTHIE** [comment]
Thing is that there are multiple keys to delete and if a module or other are using the addresses they will have to delete the cache too. If you want we can just delete the ps_customeraccount* and ps_linklist* key.

File: controllers/front/AddressController.php [hunk 1]
```diff
@@ -58,6 +58,7 @@ public function init()
      */
     public function postProcess()
     {
+    	Tools::clearAllCache();
```
Comment (PrestaEdit) [inline controllers/front/AddressController.php:61]: Indentation seems not good

File: controllers/front/AddressController.php [hunk 2]
```diff
@@ -78,7 +79,6 @@ public function postProcess()
                 } else {
                     $this->success[] = $this->trans('Address successfully added!', [], 'Shop.Notifications.Success');
                 }
-
```
Comment (PrestaEdit) [inline controllers/front/AddressController.php:81]: Removing line.

File: controllers/front/AddressController.php [hunk 3]
```diff
@@ -107,6 +107,7 @@ public function postProcess()
                 new Address($id_address, $this->context->language->id),
                 Tools::getValue('token')
             );
+
```
Comment (PrestaEdit) [inline controllers/front/AddressController.php:110]: Adding line

**Review by PrestaEdit — CHANGES_REQUESTED** [review]
Does not hesitate to make a PHP CS Fixer.

**Pull request closed without merging**